src/com/ka/spreadsheet/diff/StdoutSpreadSheetDiffCallback.java

```diff
@@ -56,6 +75,13 @@ public void reportExtraCell(boolean inFirstSpreadSheet, CellPos c) {
 
   @Override
   public void reportDiffCell(CellPos c1, CellPos c2) {
+    assert (c1.getRowIndex() == c2.getRowIndex())
```

Comment:
Since these assertions are going to repeat in every callback - we could have an Abstract class to have this logic once. Really appreciate you adding this in the current callback too.

Feel free to skip this suggestion!

Reply:
I considered that - changing `SpreadSheetDiffCallback` from an `interface` to an `abstract`.  I didn't want to mess with your structure - some folks are REALLY religious about interfaces (I worked on a product where EVERY module had an interface, and NONE had more than one implementation :-) ).  Or were you thinking something like `SpreadSheetDiffCallback` <- `SpreadSheetDiffCallbackBase` <- (`StdoutSpreadSheetDiffCallback`, `UnifiedDiffSpreadSheetDiffCallback`) ?

Reply:
Yep I was thinking interface <- abstract class <- (the 2 implementations)

Reply:
OK, I'll create a new PR for that.